spec/lib/api/v3/utilities/custom_field_injector_spec.rb

```diff
@@ -0,0 +1,326 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe ::API::V3::Utilities::CustomFieldInjector do
+  let(:cf_path) { "customField#{custom_field.id}" }
+  let(:injector) { described_class.new(modified_class) }
+  let(:field_format) { 'bool' }
+  let(:custom_field) {
+    FactoryGirl.build(:custom_field,
+                      id: 1,
+                      field_format: field_format,
+                      is_required: true)
+  }
+
+  describe 'TYPE_MAP' do
+    it 'supports all available formats' do
+      Redmine::CustomFieldFormat.available_formats.each do |format|
+        expect(described_class::TYPE_MAP[format]).to_not be_nil
+      end
+    end
+  end
+
+  describe '#inject_schema' do
+    let(:modified_class) { Class.new(::API::Decorators::Schema) }
```

Comment:
The idea of splitting up the injector into one for schema and one for actual objects does get some support when I see this, because you are actually testing both classes here. The test expects json paths to match which is a functionality provided by ::API::Decorators::Schema.

Would it make sense in your opinion to abandon the injector (the schema part) and make it a part of the Schema because it actually relies on Schema's interface rather heavily.